### ISSUE TITLE
HDF reader is very very slow.

### ISSUE THREAD

**Issue by rainwoodman** [bccp/nbodykit#283]
I created a source like this
```
source = Source.File(HDFFile, '00000400-40-wig/fof_1.0000.hdf5',
                     BoxSize=1380.0, 
                     M0 = 27.75e10 * 0.307 * 1380**3 / 2048 ** 3)
# cleaning
source['Position'] = source['Position'] * source.attrs['BoxSize']
source['Velocity'] = source['Velocity'] * source.attrs['BoxSize']
source['Mass'] = source['Length'] * source.attrs['M0']
source['Selection'] = source['Mass'] > 1e13
```

Then either computing a column or running FFTPower becomes extremely slow. The hdf5 file can be accessed from my home directory , cosmosim/vacan/00000400-40-wig/fof_1.0000.hdf5. (or any old nbodykit fof catalogues).

**Comment by rainwoodman** [comment]
@nickhand  Where does it smell bad in the reader?

**Comment by nickhand** [comment]
How large are the catalogs? I don't have access to cosmosim. If you push to an open scratch dir or m779 I can take a closer look. 

I wonder if there are many dask chunks, then the calculation of each dask chunk might be reading the full FOF catalog then slicing it...

**Comment by nickhand** [comment]
or maybe this is tied to h5py's version of fancy indexing? i.e., http://stackoverflow.com/questions/21766145/h5py-correct-way-to-slice-array-datasets

**Comment by rainwoodman** [comment]
No I don't think so. We select after obtaining the numpy arrays.

You can look at any hdf5 file from ~yfeng1/desi/baoshift/run4/*/*.hdf5 . 
Worse case we can cache these like csvfiles. I'll take a look if converting bigfile we suffer from slow down too.

**Comment by rainwoodman** [comment]
I can confirm bigfile reader is not as slow.

**Comment by rainwoodman** [comment]
closed by #284 284
